jenkins-build-pack-api/Jenkinsfile

```diff
@@ -797,10 +796,40 @@ def loadServiceInfo(service, domain, serviceUrl, endpointKey) {
 		preserveEnvKey = 'dev'
 	}
 
-	def curlOutput = sh (script: "curl -H \"Content-Type: application/json\" \
-			-H \"Authorization: $g_login_token\" \"$url\" | $jq -r '.data[0] | [.id, .endpoints[\"$preserveEnvKey\"], .created_by] | join(\",\")'", returnStdout: true)
+	def curlCMDOutput = sh (script: "curl -H \"Content-Type: application/json\" \
+			-H \"Authorization: $g_login_token\" \"$url\"", returnStdout: true)
+			
+	def jsonParser = new groovy.json.JsonSlurper()
+	def resultJson = jsonParser.parseText(curlCMDOutput)
+	def curlOutputJoin = ''
+	if (resultJson != null && resultJson.data != null )
+	{
+		if( resultJson.data[0].id != null){
+			curlOutputJoin = resultJson.data[0].id 
+		}
+		else{
+			curlOutputJoin = curlOutputJoin + "," + ""
+		}
+		
+		if( resultJson.data[0].endpoints != null && resultJson.data[0].endpoints["$preserveEnvKey"] != null){
+			curlOutputJoin = curlOutputJoin + "," + resultJson.data[0].endpoints["$preserveEnvKey"]
+		}
+		else{
+			curlOutputJoin = curlOutputJoin + "," + ""
+		}
+		
+		if( resultJson.data[0].created_by != null){
+			curlOutputJoin = curlOutputJoin + "," + resultJson.data[0].created_by
+		}
+		else{
+			curlOutputJoin = curlOutputJoin + "," + ""
+		}
+		
+
+	}
 
-	def values = curlOutput.split(',')
+	echo "curlOutputJoin:" + "$curlOutputJoin"
+  	def values = curlOutputJoin.split(',')
```

Comment:
check if curlOutputJoin is null or empty before attempting split.

Reply:
suggested changes has been made in the code